### PR TITLE
Docstrings and comments

### LINKED ISSUE #6: Docstrings
We should add docstrings to all the publicly available methods including methods in Base. For example, it would be useful to clarify what `close(deferred_channel)` is doing.

### PR DESCRIPTION
**Pull request by iamed2**
Fixes #6

### CODE REVIEW

File: src/DeferredFutures.jl [hunk 1]
```diff
@@ -58,6 +89,13 @@ function finalize_ref(ref::DeferredRemoteRef)
     return nothing
 end
 
+"""
+    reset!{T<:DeferredRemoteRef}(ref::T) -> T
+
+Removes any data from the `DeferredRemoteRef` and allows it to be reinitialized with data.
```
Comment (rofinn) [inline src/DeferredFutures.jl:105]: I'd also mention that `finalize` is called on the data.

File: src/DeferredFutures.jl [hunk 2]
```diff
@@ -58,6 +89,13 @@ function finalize_ref(ref::DeferredRemoteRef)
     return nothing
 end
```
Comment (rofinn) [inline src/DeferredFutures.jl:101]: Given that you're exporting `DeferredRemoteRef` for folks to subtype, I'd add a docstring to `finalize_ref` to explain what's going on when `finalize` is called on a `DeferredRemoteRef`.

**Review by rofinn — CHANGES_REQUESTED** [review]

File: src/DeferredFutures.jl (hunk 1 above)
Comment (iamed2) [reply src/DeferredFutures.jl:105]: For posterity: it does not

File: src/DeferredFutures.jl (hunk 2 above)
Comment (iamed2) [reply src/DeferredFutures.jl:101]: Done

File: src/DeferredFutures.jl (hunk 1 above)
Comment (rofinn) [reply src/DeferredFutures.jl:105]: Oops, yep. Sorry.

**Review by rofinn — APPROVED** [review]

**Pull request merged**